### PR TITLE
User 3 - add user login/register pages and controllers and back end

### PR DESCRIPTION
**Pull request by anupmak**
Login and registration is done for user. After login/registration user is going to website page.

@jannunzi  @MarcoAnnunziata @nischaym @hardik-surana @kethamakka 
please review this PR.

### CODE REVIEW

File: public/ide/views/home/home.controller.js [hunk 1]
```diff
@@ -6,7 +6,65 @@
...
+        vm.login = login;
+        vm.register = register;
+        vm.register = register;
+        vm.toggleNavBar = toggleNavBar;
+
+        function toggleNavBar(val) {
+            if (val == true) {
+                document.getElementById("mySidenav").style.width = "60%";
+                document.getElementById("mySidenav").style.display = "block";
+            }
```
Comment (nischaym) [inline public/ide/views/home/home.controller.js]: lets use ES6 style.
we can use '' instead of ""

File: public/ide/views/home/home.controller.js [hunk 2]
```diff
@@ -6,7 +6,65 @@
...
+            else if (user.password != user.password2) {
+                vm.alert = 'Password does not match';
+            }
+            else {
+                UserService.createUser(user)
+                    .success(function (userObj) {
+                        $rootScope.currentUser = userObj;
+                        $location.url("/user/" + userObj._id + "/website");
+                    })
```
Comment (nischaym) [inline public/ide/views/home/home.controller.js]: can we use ES6?? `${}`

File: public/ide/views/home/home.controller.js (hunk 1 above)
Comment (jannunzi) [reply public/ide/views/home/home.controller.js]: Anupma, please use ng-if or ng-show instead of changing styles from the controller.

Comment (jannunzi) [reply public/ide/views/home/home.controller.js]: OR...use ng-class="{'sixty-percent':.., 'block': ....}"

**Pull request merged**